### ISSUE TITLE
Extend keyword search to all descriptive metadata fields, not just title

### ISSUE THREAD

**Issue by newmanld** [uclibs/scholar_uc#155]
Early feedback from Early Adopters is confirming this as a high priority.  

This is already written up as a Use Case -- Submission Number 6 in https://github.com/uclibs/scholar_use_cases/blob/master/search_retrieval/search_retrieval_use_cases.md

Notre Dame appears to have done this.

**Comment by jamesvanmil** [comment]
The default query fields are being set here:

https://github.com/uclibs/curate_fork/blob/develop/app/controllers/catalog_controller.rb#L42-L53

It looks like this function is looking for a config file that we're missing in the app (`/config/search_config.yml`), and so is defaulting to search only `desc_metadata__title_tesim` and desc_metadata__name_tesim`.

Here's a generator for that file: https://github.com/uclibs/curate_fork/blob/develop/lib/generators/curate/search_config/search_config_generator.rb

**Comment by jamesvanmil** [comment]
I'm close to cracking this one but I've run into a blocker with the Curate.configuration class, which is where the default search info is stored.

In both the dummy app and in scholar I've added `/config/search_config.yml`.

This change is working in the dummy application within the gem, and from the rails console within the dummy app, invoking this class returns robust information:

```ruby
> Curate.configuration

 => #<Curate::Configuration:0x007faa621375b8 @application_root_url="http://localhost", @default_antivirus_instance=#<Proc:0x007faa5c9ef800@/Users/vanmiljf/development/gems/curate_fork/spec/internal/config/initializers/curate_config.rb:3 (lambda)>, @characterization_runner=#<Proc:0x007faa5c9ef7d8@/Users/vanmiljf/development/gems/curate_fork/spec/internal/config/initializers/curate_config.rb:6 (lambda)>, @registered_curation_concern_types=["GenericWork", "Dataset", "Article", "Image", "Document"], @search_config={"catalog"=>{"qf"=>["desc_metadata__creator_tesim", "desc_metadata__title_tesim"]}, "people"=>nil}>
```

But changes are not working as expected within the application. From the rails console within Scholar, the info is far less complete, and a message is being returned indicating that the search_config hasn't been set:

```ruby
> Curate.configuration

 => #<Curate::Configuration:0x007fba4e3a29f0 @registered_curation_concern_types=["GenericWork", "Dataset", "Article", "Image", "Document"], @application_root_url="bamboo_application_url">
 => "search_config not set"
```

Am I missing some step in getting particular config yml files to be processed?

**Comment by jamesvanmil** [comment]
I think I found what I'm missing:

The curate dummy app generator is creating a fuller /config/initializers/curate_config.rb file than what's tracked in Scholar. Compare:

```ruby
Curate.configure do |config|
        config.application_root_url = 'http://localhost'
        config.default_antivirus_instance = lambda {|file_path|
          AntiVirusScanner::NO_VIRUS_FOUND_RETURN_VALUE
        }
        config.characterization_runner = lambda {|file_path|
          Curate::Engine.root.join('spec/support/files/default_fits_output.xml').read
        }
  config.register_curation_concern :generic_work
  config.register_curation_concern :dataset
  config.register_curation_concern :article
  config.register_curation_concern :image
  config.register_curation_concern :document
  # # You can override curate's antivirus runner by configuring a lambda (or
  # # object that responds to call)
  # config.default_antivirus_instance = lambda {|filename| … }

  # # Used for constructing permanent URLs
  # config.application_root_url = 'https://repository.higher.edu/'

  # # Used to load values for constructing SOLR searches
  search_config_file = File.join(Rails.root, 'config', 'search_config.yml')
  config.search_config = YAML::load(File.open(search_config_file))[Rails.env].with_indifferent_access

  # # Override the file characterization runner that is used
  # config.characterization_runner = lambda {|filename| … }
end
```

https://github.com/uclibs/curate_fork/blob/develop/lib/generators/curate/curate_generator.rb#L72-L96

and:

```ruby
Curate.configure do |config|
  config.register_curation_concern :generic_work
  config.register_curation_concern :dataset
  config.register_curation_concern :article
  config.register_curation_concern :image
  config.register_curation_concern :document
  # # You can override curate's antivirus runner by configuring a lambda (or
  # # object that responds to call)
  # config.default_antivirus_instance = lambda {|filename| … }

  # # Used for constructing permanent URLs
config.application_root_url = 'bamboo_application_url'

  # # Override the file characterization runner that is used
  # config.characterization_runner = lambda {|filename| … }
end
```

https://github.com/uclibs/scholar_uc/blob/develop/config/initializers/curate_config.rb

**Comment by hortongn** [comment]
Merged into develop branch

**Comment by newmanld** [comment]
Fields now searchable in keyword search:

abstract
alternate_title
bibliographic_citation
contributor
coverage_spatial    <from geographic subject>
coverage_temporal   <from time period>
creator
description
genre
journal_title
language
publisher
publisher_digital
rights
subject
type
title   <title of a work but also job title on a profile>
alternate_phone_number <from profile>
blog <from profile>
campus_phone_number <from profile>
personal_webpage <from profile>
name <from profile>

**Comment by newmanld** [comment]
date_created, notes not included.  Evaluate how difficult to add these and any other omitted descriptive metadata fields.

**Comment by newmanld** [comment]
@jamesvanmil please check with metadata working group about this as well

**Comment by jamesvanmil** [comment]
@newmanld @lawhorkl all clear to proceed from Metadata WG

**Comment by lawhorkl** [comment]
@jamesvanmil @newmanld i have branches up with these fields added and working on both scholar and curate, tomorrow morning james and I will go over what we need to do to cover the specs.

**Comment by newmanld** [comment]
I am re-opening this issue because I am finding other descriptive metadata fields that are not indexed.  I have not yet checked the ETD worktype, but so far, have verified that these fields are not indexed:
Required Software
Citation
ISSN
Source (image work type)
View date (image work type) (called Date photographed on input form)
Culture (image work type) (cultural context on input form)
Geographic location (image work type) (just location on input form)
Measurements (image work type)
Material (image work type)

**Comment by scherztc** [comment]
The latest merged PR was for notes and date_created.

**Comment by newmanld** [comment]
Yep - @scherztc is correct that the PR only indexed those two fields.  But this issue says to index all descriptive fields.  Unless there are objections from metadata SMEs or others against indexing certain fields.  Thus the issue is re-opened.

**Comment by lisa3711** [comment]
I have my environment working finally. But seems like things are not working right. I've updated the files in  here:
-Curate/app/repository_datastreams/
as well as:
-scholar_uc/config/search_config.yml

I've also update my gemfile to point to local curate, for udpated files. I've edited the error for the port 25 error (though it didnt correct it, but it only happens when you create an account. )

Any other ideas?
